File: cpp/2149.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int l = nums.size();
        vector<int> res(l);
        for (int i = 0, n = 1, p = 0; i < l; i++) {
            if (nums[i] < 0) {
                res[n] = nums[i];
                n += 2;
            } else {
                res[p] = nums[i];
                p += 2;
            }
        }
        return res;
    }
};

File: python3/2149.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums)
        p = 0
        n = 1
        for i in range(len(nums)):
            if nums[i] < 0:
                res[n] = nums[i]
                n += 2
            else:
                res[p] = nums[i]
                p += 2
        return res
